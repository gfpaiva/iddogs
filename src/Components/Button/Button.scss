@import '../../Scss/root';

.button {
	@media screen and (max-width: $sm-beakpoint) { width: 100% }
	@include rfs(22);
	border: none;
	border-radius: $border-radius;
	cursor: pointer;
	display: inline-block;
	padding: 0;
	outline: none;

	&__content {
		& > *:first-child {
			display: inline-block;
			padding: .5em 1em;
		}
	}

	&:not([disabled]) { @include hover-shadow }
	&[disabled] { background-color: lighten($primary-color, 15%) }
}