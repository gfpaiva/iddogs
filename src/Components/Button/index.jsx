import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

export default function Button({
  small, className, children, ...rest
}) {
  return (
    // eslint-disable-next-line react/button-has-type
    <button
      className={`${small ? 'button--small ' : ''}${className ? `${className} ` : ''}button bg-secondary color-primary`}
      {...rest}
    >
      <span className="button__content">
        {children}
      </span>
    </button>
  );
}

Button.defaultProps = {
  small: false,
  className: '',
};

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  small: PropTypes.bool,
  className: PropTypes.string,
};
