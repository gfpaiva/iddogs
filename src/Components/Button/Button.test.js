import React from 'react';
import { mount } from 'enzyme';

import Button from './index';

describe('<Button />', () => {
  it('should mount properly', () => {
    const wrapper = mount(
      <Button>
        Test
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('shoud mount with big property', () => {
    const wrapper = mount(
      <Button small>
        Test
      </Button>
    );

    expect(wrapper.find('button').hasClass('button--small')).toBeTruthy();
  });

  it('shoud mount with custom className', () => {
    const wrapper = mount(
      <Button className="button--test">
        Test
      </Button>
    );

    expect(wrapper.find('button').hasClass('button--test')).toBeTruthy();
  });
});
