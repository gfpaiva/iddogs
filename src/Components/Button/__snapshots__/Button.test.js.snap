// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should mount properly 1`] = `
<Button
  className=""
  small={false}
>
  <button
    className="button bg-secondary color-primary"
  >
    <span
      className="button__content"
    >
      Test
    </span>
  </button>
</Button>
`;
